### DISCUSSION TITLE
Welcome to php-mime-mail-parser Discussions!

### DISCUSSION THREAD

**Discussion by eXorus** [php-mime-mail-parser/php-mime-mail-parser#356] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ravizaum** [comment]
Is it possible to directly download attachments client-side without saving them to the server?

**Reply by eXorus** [reply]
It's a backend lib but yes, you have to store the email first and after with an API you can get the stream of the attachment.
It's what I did with https://mailcare.io.

**Reply by Ravizaum** [reply]
Wow, thanks for the speedy response! 

So, if I understood that correctly, I can get the stream using your php-mime-mail-parser, and save it as a file client-side without temporary files on the server running the script (appart for the email itself, which I have stored as MIME). How can I get the stream?

Also, how could I use "$attachment->getMimePartStr()" to get the actual file base64 content? For me it's returning the content type, Id, disposition and etc AND the actual content

![Screenshot_10](https://user-images.githubusercontent.com/18106017/109587410-ce2bab80-7ae5-11eb-82c0-f9377eab38e0.png)

Thank you so much!

**Reply by eXorus** [reply]
The email stored as MIME already contains the attachment, so your already have the attachments stored physically on your server.

On mailcare I do something like that:
https://github.com/mailcare/mailcare/blob/master/app/Http/Controllers/AttachmentsController.php

It's an API that return the attachment with the correct content type to be well understood by my front-end.
```php
 $data = stream_get_contents($attachmentParsed->getStream());
 return response($data)->header('Content-Type', $attachmentParsed->getContentType());
```

**Reply by Ravizaum** [reply]
Thank you very much!

**Comment by kilixn** [comment]
Hey there  ,

I just found this mail parser and want to try it out for some testing.
Is there any way to connect via IMAP instead of the other ones (files etc.)?

Sorry if this is obvious, I'm new...

Thanks in advance!

PS : I already have the whole app I just need the parser so can you guys recommend the best way? (using IMAP)

**Reply by eXorus** [reply]
There are already methods in PHP to connect to imap and imap already provide a parsing so you don’t need this lib for this use case.

https://www.php.net/manual/en/ref.imap.php

or you can check out this link:
https://github.com/php-mime-mail-parser/php-mime-mail-parser/issues/262

**Reply by kilixn** [reply]
I'm already using the IMAP extension but the decoding with the HTML Emails won't work properly so I thought I need an Mail parser or so... 
Thanks, I'm going to try the mail parser. But which parsing do you recommend with imap library?

Thanks again for the Help !

**Reply by eXorus** [reply]
So check out the issue #262 to use this library with the imap extension
